Accound card re-design and few related tasks

tigranpetrossian opened the pull request:
Closes LEA-1912, LEA-1910, LEA-1914
Partially addresses LEA-1909

Updates account card to [the new design](https://www.figma.com/design/FlmF7ropcijOb7iVSDqNn0/%F0%9F%9F%AA-App-Explorations-2024-H2?node-id=11422-78213&t=fuLCxkUs6knrn7c1-4), and includes few adjustments in related views.

### Account card
* Updated [account card](https://github.com/leather-io/mono/blob/086220a3304bbe3fb1e7f1e8bf0f3959a2005449/apps/mobile/src/features/accounts/components/account-card.tsx) design
* Account widget items now use the same account card as it's variation of the same component by design.
* Updated account avatars to use the new squircle design, renamed to AccountAvatar

#### Account widgets

https://github.com/user-attachments/assets/290084dd-f49d-4929-9b27-93772e568d90

#### Account sheet, also fixed header positioning.

<img width="320" alt="account sheet" src="https://github.com/user-attachments/assets/9f98b974-881b-4a0e-af89-7b9a9666e820" />

#### Settings - Wallets screen

<img width="320" alt="wallets-screen" src="https://github.com/user-attachments/assets/00920140-7cbd-4d06-b8f0-af28eb727e7a" />

#### Account listing

<img width="320" alt="account-listing" src="https://github.com/user-attachments/assets/66536baa-3e15-4c87-a9f7-3015d26604e9" />

#### Account overview
<img width="421" alt="account-overview" src="https://github.com/user-attachments/assets/2c5c1eef-7ad4-4cf8-b7df-1cd05ffdcf36" />

### Additional updates

#### Account sheet no longer spans behind the unsafe area

https://github.com/user-attachments/assets/28f0e92f-f270-42f1-996d-199d17f368e7

#### Updated Choose Avatar screen, added icon preview.

<img width="320" alt="choose-avatar" src="https://github.com/user-attachments/assets/bca45461-bb38-491b-92c7-64ad13eb4f64" />


### Few implementation notes
* New AccountAvatar will likely build on Avatar component from the UI library. For now it's its own component, as I didn't want to Avatar/Icon refactor to block this for too long.
* The Account card component warrants another refactor apart from re-design, to contain its own components vs. having them injected via props, as it's a fairly rigid component with very few variations. I de-scoped this, but would love to pair with someone and look at standardizing account entity models throughout the app as a prerequisite for doing this.
* Almost unrelated: removed tap-to-toggle functionality from PrivateText, as it wasn't being used, part of this PR as it was interfering with Balance positioning. Will re-visit as necessary.

tigranpetrossian commented:
@mica000 @markmhendrickson @fabric-8 Demo build should be up for previewing.

pete-watters commented on apps/mobile/src/components/private-text.tsx:
```diff
@@ -1,37 +1,13 @@
 import { Text, TextProps } from '@leather.io/ui/native';
 
-const defaultPrivateText = '***';
+const defaultMask = '***';
```
This single change is an awesome refactor 👍

pete-watters commented on apps/mobile/src/features/accounts/account-selector/account-selector-sheet-header.tsx:
```diff
@@ -31,7 +31,7 @@ export function AccountSelectorHeader({ sheetRef }: AccountSelectorHeaderProps)
       <Box alignItems="flex-end" flexGrow={1} justifyContent="center" zIndex="20">
         <Pressable
-          p="2"
+          hitSlop={12}
```
Nice 👍

pete-watters commented on apps/mobile/package.json:
```diff
@@ -91,6 +91,7 @@
     "expo-router": "3.5.14",
     "expo-secure-store": "13.0.2",
     "expo-splash-screen": "0.27.4",
+    "expo-squircle-view": "1.1.0",
```
Do we need this in the mobile app + the UI package or can we just have it in the UI package and consume that component?

tigranpetrossian replied:
Good point, looking at other native packages in the UI this might be unnecessary.

pete-watters reviewed (approved):
Great work on this @tigranpetrossian 🚀 

I think there may be an un-needed import of `expo-squircle-view` but I could be mistaken. 

It's flawless otherwise with some nice renaming, refactors + UX improvements

kyranjamie commented on apps/mobile/src/features/accounts/account-selector/account-selector-sheet.layout.tsx:
```diff
@@ -31,62 +31,61 @@ export function AccountSelectorSheetLayout({
+                onCardPress={() => onAccountPress(account.id)}
+                swapCardIndexes={swapAccountIndexes}
+              >
+                <WalletLoader fingerprint={account.fingerprint} key={account.id}>
+                  {wallet => (
+                    <AccountCard
+                      caption={wallet.name}
+                      primaryTitle={account.name}
+                      secondaryTitle={
+                        <AccountBalance
+                          variant="label01"
+                          accountIndex={account.accountIndex}
+                          fingerprint={account.fingerprint}
+                        />
+                      }
+                      address={
+                        <AccountAddress
+                          accountIndex={account.accountIndex}
+                          fingerprint={account.fingerprint}
+                        />
+                      }
+                      icon={account.icon}
+                      iconTestID={defaultIconTestId(account.icon)}
+                    />
+                  )}
+                </WalletLoader>
```
I see this was here before, but isn't depending on a WalletLoader (and thus state), beyond the responsibility of a layout component?

Shouldn't everything within the loader be passed from the parent?

kyranjamie commented on apps/mobile/src/features/accounts/components/account-card.tsx:
```diff
@@ -1,68 +1,92 @@
+    return null;
+  }
+
+  return isString(children) ? (
```
slight preference for early return, more readable and consistent with above return

kyranjamie reviewed (approved):
Solid work @tigranpetrossian 🦾

fbwoolf reviewed (approved):
Nice work @tigranpetrossian 💪

mica000 commented:
This is super cool to see @tigranpetrossian! Thank you so much for the quick work on this and for the eagle on details! 

There seems to exist a container around the AccountWidget that is "cutting" the shadow. Could you check if this is true?
![CleanShot 2025-01-14 at 15 03 03@2x](https://github.com/user-attachments/assets/808bcf12-5d15-423f-8bbd-8faaa2161a08)

On smaller devices this happening. This its relates to the grid and spacing between the icons?
![CleanShot 2025-01-14 at 11 16 38@2x](https://github.com/user-attachments/assets/e92fc261-629a-4069-af4d-fcee1ab54506)

Small design request: Would be possible to use border-default to color the border of the AvatarIcon, instead of border-transparent, as initially requested? Im seeing that is a bit strange to use border-transparent when all the rest is brown (mental note to only use border-transparent when there is a background image or a color as a fill)

mica000 commented:
### Smaller devices
On smaller devices this happening. This its relates to the grid and spacing between the icons?
![CleanShot 2025-01-14 at 11 16 38@2x](https://github.com/user-attachments/assets/e92fc261-629a-4069-af4d-fcee1ab54506)

mica000 commented:
### Small design request
Would be possible to use border-default to color the border of the AvatarIcon, instead of border-transparent, as initially requested? Im seeing that is a bit strange to use border-transparent when all the rest is brown (mental note to only use border-transparent when there is a background image or a color as a fill)

![CleanShot 2025-01-14 at 15 17 27@2x](https://github.com/user-attachments/assets/8989b2b2-5418-462d-b7b2-ee431542c593)

mica000 commented:
@fabric-8 smt to consider about the copy since the card for accounts got an update and the title and subtitles are now inverted the hierarchy. Can we just keep "Create, Import or connect instantly" and remove "Create or restore wallet" or vice and versa? Also curious about what connect means in here.. ?

![CleanShot 2025-01-14 at 17 04 18@2x](https://github.com/user-attachments/assets/9a0d63ec-1275-4966-915a-dfbebef5037c)

mica000 commented:
@tigranpetrossian I noticed you mentioned the header being fixed, but I’m not sure if you’re referring to it being stuck at the top. Currently, the header scrolls along with the page, but ideally, it should remain sticky.

![CleanShot 2025-01-14 at 17 16 42](https://github.com/user-attachments/assets/bf730704-d49d-4f31-b853-8422e395c5df)

tigranpetrossian commented:
@mica000 Really appreciate all the feedback.

> There seems to exist a container around the AccountWidget that is "cutting" the shadow. Could you check if this is true?

True! Fixed.

> Would be possible to use border-default to color the border of the AvatarIcon, instead of border-transparent

Of course. Changed.

> On smaller devices this happening. This its relates to the grid and spacing between the icons?

Good catch. Thanks for stress testing this! I'll be tracking this separately [here](https://linear.app/leather-io/issue/LEA-1985/account-avatar-grid-alignment-issues), and will get to it shortly.

> I noticed you mentioned the header being fixed, but I’m not sure if you’re referring to it being stuck at the top. Currently, the header scrolls along with the page, but ideally, it should remain sticky.

I was referring to adjusting the positioning of the header and the gear icon, as it looked off compared to designs. Additionally there was a bug with the sheet expanding beyond the safe area at the top which was fixed in this PR which accidentally addressed [this issue](https://linear.app/leather-io/issue/LEA-1914/set-a-max-height-for-account-selector-drawer).

Agreed on sticky headers, made an [issue](https://linear.app/leather-io/issue/LEA-1986/make-the-header-in-account-sheet-sticky).

> @fabric-8 smt to consider about the copy since the card for accounts got an update and the title and subtitles are now inverted. Can we just keep "Create, Import or connect instantly" and remove "Create or restore wallet" or vice and versa? Also curious about what connect means in here.. ?

👍 Keep me posted and I'll update the copy.

tigranpetrossian commented:
@mica000 Merging this to avoid it hanging here for too long. Unaddressed feedback is tracked in Linear. Will update the copy separately as needed.

markmhendrickson commented:
Very nice work, @tigranpetrossian 💯  I started QA'ing and liked a lot what I saw (...and agree about merging this in before I was finished since I shouldn't be a blocker generally).

tigranpetrossian commented:
> Very nice work, @tigranpetrossian 💯 I started QA'ing and liked a lot what I saw (...and agree about merging this in before I was finished since I shouldn't be a blocker generally).

Thanks @markmhendrickson! Please send any feedback my way, I'll be doing at least one iteration on this anyway.
